src/shared/contexts/appReducer.ts

```diff
@@ -0,0 +1,33 @@
+import { Action, Type } from './actions';
+
+import storage from '../utils/storage';
+export const initialAppState = storage.get('user');
+export type Appstate = typeof initialAppState;
+
+export type ReducerActions = Action;
+
+export function appReducer(state: Appstate, action: Action) {
+  const { type, payload } = action;
+  switch (type) {
+    case Type.LOGIN:
+      storage.set('user', JSON.stringify(payload));
```

Comment:
Reducers should be pure functions, side effects don't belong in here. Instead you should expose a `login` function that dispatched to update the state and also manages local storage